lib/rspec/mocks/method_double.rb

```diff
@@ -63,6 +63,7 @@ def define_proxy_method
           define_method(method_name) do |*args, &block|
             method_double.proxy_method_invoked(self, *args, &block)
           end
+          ruby2_keywords(method_name) if respond_to?(:ruby2_keywords, true)
```

Comment:
Aha! Thats where its needed.

Reply:
@bryanp 👏 

Just wondering how did you figure it out? 2.7.1's warning traces are kind of unhelpful.

Reply:
@pirj I found a test case that raised an exception, so there was a pretty strong hint in the backtrace:

```
     ArgumentError:
       wrong number of arguments (given 1, expected 0; required keyword: bar)
     # ./spec/rspec/mocks/and_call_original_spec.rb:189:in `foo'
     # ./lib/rspec/mocks/message_expectation.rb:101:in `call'
     # ./lib/rspec/mocks/message_expectation.rb:101:in `block in and_call_original'
     # ./lib/rspec/mocks/message_expectation.rb:740:in `call'
     # ./lib/rspec/mocks/message_expectation.rb:572:in `invoke_incrementing_actual_calls_by'
     # ./lib/rspec/mocks/message_expectation.rb:427:in `invoke'
     # ./lib/rspec/mocks/proxy.rb:217:in `message_received'
     # ./lib/rspec/mocks/proxy.rb:361:in `message_received'
     # ./lib/rspec/mocks/method_double.rb:78:in `proxy_method_invoked'
     # ./lib/rspec/mocks/method_double.rb:64:in `block (2 levels) in define_proxy_method'
     # ./spec/rspec/mocks/and_call_original_spec.rb:191:in `block (3 levels) in <top (required)>'
```

Reply:
Ah yes, sadly 2.7 is less helpful

Reply:
Sorry for the inconvenience 🙏
Just a tip: if you run it on Ruby 3.0, it will raise an exception instead of a warning, which will show a more helpful backtrace.